src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -107,20 +107,22 @@ class ConfigurationHelper implements Serializable {
         return false
     }
 
-    def getMandatoryProperty(key, defaultValue = null) {
+    def getMandatoryProperty(key, defaultValue = null, errorMessage = null) {
 
         def paramValue = config[key]
 
         if (paramValue == null)
             paramValue = defaultValue
 
-        if (paramValue == null)
-            throw new IllegalArgumentException("ERROR - NO VALUE AVAILABLE FOR ${key}")
+        if (paramValue == null) {
+            if(! errorMessage) errorMessage = "ERROR - NO VALUE AVAILABLE FOR ${key}"
+            throw new IllegalArgumentException(errorMessage)
```

Comment:
I would suggest to make use of the SimpleTemplateEngine:
```
import groovy.text.SimpleTemplateEngine

def getMandatoryProperty(key, defaultValue = null, errorMessage = "ERROR - NO VALUE AVAILABLE FOR ${key}"){
  [...]
  errorMessage = SimpleTemplateEngine.newInstance()
    .createTemplate(errorMessage)
    .make([key: key]).toString()

  throw new IllegalArgumentException(errorMessage)
```

Reply:
@CCFenner mmmhhhhhh ... Sounds like shooting on sparrows with cannons. Should we do that all the time we create a string containing a parameter? We have thousands of similar statements in the code using groovy strings for that. What is bad with the "good old" groovy string approach? Would make sense to use a template engine for big strings, like html templates where we have to replace variables with values when performance is an issue. But for a very simple and short string. What would be your general recommendation for using a TemplateEngine vs. using the groovy string pattern?

Reply:
You are right, the `key` is handed in from the outside anyway...